src/views/AnalysisView.vue

```diff
@@ -99,10 +102,10 @@ const matrix = computed(() => {
       <table>
         <tr>
           <th></th>
-          <th v-for="i of Math.min(scale.size, MAX_SCALE_SIZE)" :key="i">
+          <th v-for="i of Math.min(scale.size-1+matrixStartIndex, MAX_SCALE_SIZE-1)" :key="i">
```

Comment:
`MAX_SCALE_SIZE` is a sanity limit that makes sure the browser doesn't go :boom: . You could rename it to something like `MAX_NUM_COLUMNS` or something. That `-1` should be there on a constant.